src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubSCMSource.java

```diff
@@ -371,14 +405,14 @@ public GitHubSCMSource(@CheckForNull String id, @CheckForNull String apiUri,
                            @NonNull String checkoutCredentialsId,
                            @CheckForNull String scanCredentialsId, @NonNull String repoOwner,
                            @NonNull String repository) {
-        this(repoOwner, repository, null, false);
+        this(repoOwner, repository);
```

Comment:
The question here is if a with `@Deprecated` annotated constructor should be used (even if this is a very valid one) or just the default one + setters. Otherwise code checkers / IDEs might complain about this fact.